v3/src/components/graph/adornments/movable-line/movable-line-adornment-component.tsx

```diff
@@ -90,83 +110,93 @@ export const MovableLineAdornment = observer(function MovableLineAdornment(props
       equation.style("left", `${screenX}px`)
         .style("top", `${screenY}px`)
     } else {
-      const left = equationCoords.x * 100
-      const top = equationCoords.y * 100
-      equation.style("left", `${left}%`)
-              .style("top", `${top}%`)
+      const left = equationCoords.x
+      const top = equationCoords.y
+      equation.style("left", `${left}px`)
+              .style("top", `${top}px`)
     }
-  }, [equationContainerSelector, instanceKey, model.lines, plotHeight, plotWidth, xAttrName,
-      xScale, xSubAxesCount, yAttrName, yScale, ySubAxesCount])
+  }, [equationContainerSelector, instanceKey, model.lines, plotHeight, plotWidth, xAttrName, xSubAxesCount,
+      yAttrName, ySubAxesCount])
+
+  const breakPointCoords = useCallback((pixelPtsOnAxes: IPointsOnAxes, breakPointNum: number) => {
+    if (interceptLocked) return {x: xScaleRef.current(0), y: yScaleRef.current(0)}
+    const weight = breakPointNum === 1 ? 3/8 : 5/8
+    const x = pixelPtsOnAxes.pt1.x + weight * (pixelPtsOnAxes.pt2.x - pixelPtsOnAxes.pt1.x)
+    const y = pixelPtsOnAxes.pt1.y + weight * (pixelPtsOnAxes.pt2.y - pixelPtsOnAxes.pt1.y)
+    return {x, y}
+  }, [interceptLocked])
```

Comment:
If `interceptLocked` were passed in as an argument then this could be a utility function outside the component that didn't require `useCallback` at all.

Reply:
I moved this to graph-utils.ts. Doing so required also passing `xScaleRef.current` and `yScaleRef.current`. Does that seem reasonable? Or am I maybe missing a way that  wouldn't be necessary? I also had to move the `IPointsOnAxes` interface (to graph-utils.ts), although I'm wondering now if there's a better place for that. Maybe graphing-types.ts?

Reply:
I hadn't considered the scale refs. Given that the function is only used in the movable line, I'd be inclined to leave it where it was as a callback but to pass `interceptLocked` as an argument so that the callback is stable and won't invalidate any effects that use it. Passing it as an argument leaves it up to the caller how to handle changes to `interceptLocked`.